test/parallel/test-assert.js

```diff
@@ -409,6 +409,10 @@ assert.throws(() => {
 }, /Got unwanted exception: user message/,
               'a.doesNotThrow ignores user message');
 
+assert.throws(() => {
+  assert.doesNotThrow(makeBlock(thrower, Error), 'user message');
+}, /Got unwanted exception: user message\ntest/);
```

Comment:
Can we add a test to confirm we don't get unhelpful strings (like "undefined") if the user leaves out the optional `message` argument?